src/core/util/options.js

```diff
@@ -48,7 +49,14 @@ if (process.env.NODE_ENV !== 'production') {
 function mergeData (to: Object, from: ?Object): Object {
   if (!from) return to
   let key, toVal, fromVal
-  const keys = Object.keys(from)
+
+  const keys = hasSymbol
+    ? Reflect.ownKeys(from).filter(key => {
+      /* istanbul ignore next */
```

Comment:
Wasn't entirely sure if this was needed but took it from `inject.js` to be safe.